components/Game.tsx

```diff
@@ -192,7 +200,43 @@ export default class Game extends React.Component<{}, GameState> {
         
         return data;
     }
-    
+
+    dropDown = (data: Array<Square>) => {
+        console.log("drop down func called");
+        let i: number;
+        let type: number;
+        
+        for (i = (row * column) -1; i >= 0; i--) {
+            if (data[i].type == '0' && i > column -1) {
```

Comment:
don't hardcode the type of the cell you're checking, store these in a cell types file or an enum file, and check against that.